### DISCUSSION TITLE
UE5 Android Packaging Error - fatal error: 'FGFDMExec.h' file not found

### DISCUSSION THREAD

**Discussion by Kayzield909** [JSBSim-Team/jsbsim#699] [Q&A]
Hi, I'm trying to package the UE5 plugin for android and am getting the following error:
Branches/5.05_JSBSIM/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Private/JSBSimMovementComponent.cpp(16,10): fatal error**: 'FGFDMExec.h' file not found**

If I go into JSBSim.Build.cs and add an **else if** (Target.platform == UnrealTargetPlatform.Android) the FGFDMExec.h error goes away but now I have tonnes of new errors [Pasted below].

Please help me understand what these errors mean, if they are fixable and how long they would take a mid level programmer to fix?

Cheers

Second Errors:
________________

Plugins/JSBSimFlightDynamicsModel/Source/ThirdParty/JSBSim/Include\simgear/props/props.hxx(1893,40): error: use of typeid requires -frtti
UATHelper: Packaging (Android (ASTC)):     static const char* name() { return typeid(T).name(); }

Plugins/JSBSimFlightDynamicsModel/Source/ThirdParty/JSBSim/Include\input_output/FGOutputType.h(151,8): error: 'JSBSim::FGOutputType::Run' hides overloaded virtual function [-Werror,-Woverloaded-virtual]
UATHelper: Packaging (Android (ASTC)):   bool Run(void);
UATHelper: Packaging (Android (ASTC)):        ^

Plugins/JSBSimFlightDynamicsModel/Source/ThirdParty/JSBSim/Include\simgear/misc/sg_path.hxx(79,14): note: candidate constructor not viable: no known conversion from 'const TCHAR *' (aka 'const char16_t *') to 'const std::string' (aka 'const basic_string<char, char_traits<char>, allocator<char> >') for 1st argument
UATHelper: Packaging (Android (ASTC)):     explicit SGPath( const std::string& p, PermissionChecker validator = NULL );

Plugins/JSBSimFlightDynamicsModel/Source/ThirdParty/JSBSim/Include\simgear/misc/sg_path.hxx(88,5): note: candidate constructor not viable: requires at least 2 arguments, but 1 was provided
UATHelper: Packaging (Android (ASTC)):     SGPath( const SGPath& p,
UATHelper: Packaging (Android (ASTC)):     ^

**Comment by bcoconni** [comment]
@gallonmate, @AlbanBERGERET-Epic any idea about how this could be fixed ?

**Comment by AlbanBERGERET-Epic** [comment]
Hi, 

So far, the UE5 plugin is meant to run only on Windows/x64 platforms. 

In order to have it running on Android, you would have to first build JSBSim for the Android Target, and store the relevant libraries in the plugin subfolder (eg : jsbsim\UnrealEngine\Plugins\JSBSimFlightDynamicsModel\Source\ThirdParty\JSBSim\LibAndroid). 

But personally, I don't know if it's possible to build JSBSim for Android... 

Then, as you noticed, you'll need to edit the JSBSim.Build.cs file to add a specific target for Android, so that it links against the JSBSim libraries for Android built previously. 
The Reference app should be able to be packaged for Android, since the GeoReferencing plugin is also available for this platform. 
You can look at "\Engine\Plugins\Runtime\GeoReferencing\Source\ThirdParty\vcpkg.sh" to see how we built Proj for Android, but the process might be different with JSBSim... 

Alban
